model/apps/rmcat-receiver.cc

```diff
@@ -66,41 +76,43 @@ void RmcatReceiver::RecvPacket (Ptr<Socket> socket)
         return;
     }
 
-    Address remoteAddr;
+    Address remoteAddr{};
     auto packet = m_socket->RecvFrom (remoteAddr);
     NS_ASSERT (packet);
-    RtpHeader header;
+    RtpHeader header{};
     NS_LOG_INFO ("RmcatReceiver::RecvPacket, " << packet->ToString ());
     packet->RemoveHeader (header);
     auto srcIp = InetSocketAddress::ConvertFrom (remoteAddr).GetIpv4 ();
     auto srcPort = InetSocketAddress::ConvertFrom (remoteAddr).GetPort ();
     if (m_waiting) {
         m_waiting = false;
-        m_srcId = header.m_ssrc;
+        m_remoteSsrc = header.GetSsrc ();
         m_srcIp = srcIp;
         m_srcPort = srcPort;
     } else {
         // Only one flow supported
-        NS_ASSERT (m_srcId == header.m_ssrc);
+        NS_ASSERT (m_remoteSsrc == header.GetSsrc ());
         NS_ASSERT (m_srcIp == srcIp);
         NS_ASSERT (m_srcPort == srcPort);
     }
 
-    //TODO (deferred): We need to aggregate feedback information
-    //                 (for the moment, one feedback packet per media packet)
-
-    auto recvTimestamp = Simulator::Now ().GetMilliSeconds ();
-    SendFeedback (header.m_sequence, recvTimestamp);
+    uint64_t recvTimestampUs = Simulator::Now ().GetMicroSeconds ();
+    SendFeedback (header.GetSequence (), recvTimestampUs);
 }
 
 void RmcatReceiver::SendFeedback (uint16_t sequence,
-                                  uint64_t recvTimestamp)
+                                  uint64_t recvTimestampUs)
 {
-    FeedbackHeader header;
-    header.flow_id = m_srcId;
-    header.sequence = uint32_t (sequence); //TODO: cast to uint32_t as we're still using old feedback header
-    header.receive_tstmp = recvTimestamp;
-
+    // TODO (next patch): We need to aggregate feedback information
+    //                    (for the moment, one feedback packet per media packet)
+    //                     - add member of type feedback header
+    //                     - add timeout (100 ms), upon timeout send header
+    //                     - if TOO_LONG, send immediately
+
+    CCFeedbackHeader header{};
+    header.SetSendSsrc (m_ssrc);
+    auto res = header.AddFeedback (m_remoteSsrc, sequence, recvTimestampUs);
```

Comment:
are we maintaining and feedback all the history?

Reply:
Not sure I understood. CCFeedbackHeader class keeps all feedback items added since creation. Then, upon serialization, the header is rendered, and the class's destructor is called.
Next, a new fresh instance of CCFeedbackHeader is created and the cycle repeats.
Does this answer your question?